ql/termstructures/yield/piecewisezerospreadedtermstructure.hpp

```diff
@@ -100,17 +97,10 @@ namespace QuantLib {
                                                            Frequency freq,
                                                            DayCounter dc,
                                                            const T& factory)
-    : originalCurve_(std::move(h)), spreads_(std::move(spreads)), dates_(dates),
-      times_(dates.size()), spreadValues_(dates.size()), comp_(comp), freq_(freq),
-      dc_(std::move(dc)), factory_(factory) {
-        QL_REQUIRE(!spreads_.empty(), "no spreads given");
-        QL_REQUIRE(spreads_.size() == dates_.size(),
-                   "spread and date vector have different sizes");
+    : InterpolatedPiecewiseSpreadCurve<T>(spreads, dates, h->referenceDate(), h->dayCounter(), factory),
```

Comment:
This is a bit of a problem.  The old implementation allowed the passed `Handle<YieldTermStructure>` to be empty and to be linked later; this requires it to be already linked.  Also, if we pass a linked handle and later we relink it to a term structure with a different reference date or day counter, these won't be updated.

It's probably simpler to remove `InterpolatedPiecewiseSpreadCurve` and keep the duplication.

Reply:
No problem.  The library is used in a lot of ways, and not all of them are obvious...